lib/v2/screens/receive_screens/receive_enter_data/interactor/receive_enter_data_bloc.dart

```diff
@@ -22,21 +26,30 @@ class ReceiveEnterDataBloc extends Bloc<ReceiveEnterDataEvents, ReceiveEnterData
     if (event is OnAmountChange) {
       double parsedQuantity = double.tryParse(event.amountChanged) ?? 0;
 
+      String seedsToFiat =
+          state.ratesState.fromSeedsToFiat(parsedQuantity, settingsStorage.selectedFiatCurrency).fiatFormatted;
+
       if (parsedQuantity > 0) {
-        yield state.copyWith(isNextButtonEnabled: true);
+        yield state.copyWith(isNextButtonEnabled: true, quantity: parsedQuantity, fiatAmount: seedsToFiat);
       } else {
-        yield state.copyWith(isNextButtonEnabled: false);
+        yield state.copyWith(isNextButtonEnabled: false, quantity: parsedQuantity, fiatAmount: seedsToFiat);
       }
     }
     if (event is OnDescriptionChange) {
       yield state.copyWith(description: event.description);
     }
     if (event is OnNextButtonTapped) {
-      //Create Receive
-      //if success
-      yield state.copyWith(pageCommand: NavigateToReceiveDetails());
-      //if failure
-      yield state.copyWith(pageCommand: ShowTransactionFail());
+      yield state.copyWith(pageState: PageState.loading);
+      Result result = await ReceiveSeedsInvoiceUseCase().run(amount: state.quantity, memo: state.description);
+      yield CreateInvoiceResultMapper().mapResultToState(state, result);
+    }
+    if (event is ClearPageState) {
```

Comment:
Maybe ClearReciveEnterDataState instead ClearPageState.
This way, when we see the print in the console, We know from which place its state was cleaned.

Reply:
i dont know why we need this state here? 
I think we should not allow the user to hi back on the QR code page. 

Only the done button should be there. 
And we should handle device back to take the user to the same place the "done" button does

Reply:
Hi, I spoke with Json and he wanted the back bottom navigate to receive seeds screen and the Done to main page.